src/main/java/com/google/step/servlets/DealPostServlet.java

```diff
@@ -81,8 +93,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       return;
     }
 
-    // TODO get the tag names from request parameter
-    List<String> tagNames = new ArrayList<>();
+    String tagParameter = request.getParameter("tags");
+    if (tagParameter == null) {
+      tagParameter = "";
+    }
+    List<String> tagNames =
+        (tagParameter.isEmpty()) ? new ArrayList<>() : Arrays.asList(tagParameter.split(","));
```

Comment:
```suggestion
    List<String> tagNames =
        tagParameter == null ? new ArrayList<>() : Arrays.asList(tagParameter.split(","));
```

Reply:
but when the user makes a request with `tagParamter == ""`, then `tagParameter.split(",")` will return `[""]`, but I want an empty array instead.

Reply:
In that case, this would be clearer:

```java
List<String> tagNames = new ArrayList<>();
if (tagParameter != null && tagParameter.isNotEmpty()) {
  tagNames = Arrays.asList(tagParameter.split(","));
}
```